### ISSUE TITLE
Invalid type exception when casting empty array

### ISSUE THREAD

**Issue by kminek** [spatie/data-transfer-object#47]
Following code shoud not throw an exception:

```php
class ParentDto extends DataTransferObject
{
    /**
     * @var \ChildDto
     */
    public $child;
}

class ChildDto extends DataTransferObject
{
}

$child = new ChildDto([]);

$parent = new ParentDto([
    'child' => $child,
]);

dump($parent->toArray());

$parentFromArray = new ParentDto([
    'child' => [],
]);
```

**Comment by brendt** [comment]
@TBPixel could this be an oversight from #45 ? If so: would you like to take a look at it? Don't feel pressured! If you're not up for it, I'll gladly fix it.

**Comment by TBPixel** [comment]
Hey @brendt I'd be glad to take this on. You're right that it does seem like an oversight from #45. It looks like a simple fix.

I'll write a test to re-create the issue and then implement a fix.

**Comment by brendt** [comment]
Fixed in 1.7.1: https://github.com/spatie/data-transfer-object/releases/tag/1.7.1
